examples/Time-Series/time_series_keithley2600.py

```diff
@@ -0,0 +1,146 @@
+# Connect keithley via RS-232 to USB connector to computer and check if it is connected:
+list_resources() # -> this will list your connections e.g. ('ASRL/dev/ttyUSB1::INSTR',)
+
+# %% For interactive use connect to the instrument
+smu = Keithley2600("ASRL/dev/ttyUSB0::INSTR")
+print(smu.id)
+
+# %% Define measureement parameter
+voltage = 10 # bias in V
+duration = 5 # s
+nplc = 10 # Number power line cycles from 0.001 to 25 (f = 50 Hz or 60 Hz -> nplc = 1 / f )
+line_frequency = 50  # can we read this from the instrument?
+t_step = nplc / line_frequency  
+
+n_datapoints = int(duration / t_step + 1) # one point more since we want to measure at cero too
+i_range = 1e-3 # range of the current
+i_compliance=0.1 # max current / compliance
+
+# %% When the with-block is exited, the shutdown method of the instrument will be called, turning the system into a safe state.
+with Keithley2600("ASRL/dev/ttyUSB1::INSTR") as smu:
+
+    # First define input parameters
+    smu.reset()
+    smu.ChA.wires_mode = "2"
+    smu.ChA.source_voltage = voltage
+    sleep(0.1) # wait here to give the instrument time to react
+
+    smu.ChA.measure_current(nplc=nplc, current=i_range, auto_range=False) # with auto_range=True it takes very long and sleep(X) might to be increased
+    sleep(0.1)
+
+    # Apply bias
+    smu.ChA.apply_voltage(compliance_current=i_compliance)  # Somehow I would expect this to turn source on but it does not
+    sleep(0.1)
+    smu.ChA.source_output = "ON" # needed to turn source on
+
+    # Measure current
+    currents = np.zeros(n_datapoints)
+    times = np.zeros(n_datapoints)
+
+    for i in range(n_datapoints):
+        times[i] = time()
+        currents[i] = smu.ChA.current
+    sleep(0.1)
+    smu.ChA.source_output = "OFF" # needed to turn source off even inside with statement
+
+    # Beep once the measurement is done -> this is not implemented so we send lua strings to the smu -> for a list of commands see https://www.tek.com/keithley-source-measure-units/smu-2600b-series-sourcemeter-manual-8
+    smu.write("beeper.enable = beeper.ON")
+    smu.write("beeper.beep(.5,292.00)")
+
+    # so we actually could read the line-frequency from the device:
+    linefreq = float(smu.ask("print(localnode.linefreq)"))
+
+# %% Save the data columns in a pandas DataFrame and inspect 
+data = pd.DataFrame({
+    'Time (s)': times - times[0],
+    'Current (A)': currents
+})
+
+# For 1 MOhm and 10V bias we expect ~ 0.01 mA -> in my case it measured 0.009 mA but this should be tolerable. If your measurement is totally off check all the connections
+print(data.tail())
+print("Compare the difference of duration = {}s and the end time of {}s in our DataFrame".format(duration, data["Time (s)"][-1:].values))
+print("The timing is not accurate since there is a delaye from the for-loop. If you do only care about the number of datapoints this is not a problem")
+
+# %% Advanced example with buffer
```

Comment:
you should probably introduce a boolean selector at the beginning, that runs either one or the other example approach, so that users can easily control that and don't have both things running after another.

Reply:
Also, you can reduce duplicated code as e.g. the postprocessing is independent of which way you collect the data.